sys\net\gnrc\link_layer\lwmac: cast error with avg-gcc

#### Description
Cast error in gnrc_lwmac while build test/gnrc_lwmac for AVR-based derfmega256 board. 
Similar to #12853.
#### Steps to reproduce the issue
- Take RIOT-OS latest master
- Update the file tests/gnrc_lwmac/Makefile :  
`-BOARD_WHITELIST := samr21-xpro`
`+BOARD_WHITELIST := samr21-xpro derfmega256` 
- Execute `BOARD=derfmega256 make all`

#### Expected results
The tests/gnrc_lwmac builds on deRFmega256 without errors.

#### Actual results
cast error:
`/RIOT/sys/net/gnrc/link_layer/lwmac/lwmac.c: In function ‘rtt_cb’:
/RIOT/sys/net/gnrc/link_layer/lwmac/lwmac.c:679:26: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]`
`     msg.content.value = ((uint32_t) arg) & 0xffff;`
`                          ^`
`cc1: all warnings being treated as errors`


#### Versions
RIOT latest master

--- a/sys/net/gnrc/link_layer/lwmac/lwmac.c
+++ b/sys/net/gnrc/link_layer/lwmac/lwmac.c
@@ -676,7 +676,7 @@ static void rtt_cb(void *arg)
 {
     msg_t msg;
 
-    msg.content.value = ((uint32_t) arg) & 0xffff;
+    msg.content.value = (uint16_t)((uintptr_t) arg);
     msg.type = GNRC_LWMAC_EVENT_RTT_TYPE;
     msg_send(&msg, lwmac_pid);
 

